libmamba/src/core/query.cpp

```diff
@@ -487,24 +590,37 @@ namespace mamba
         };
 
         printers::Table printer(headers);
+        printer.set_alignment(alignments);
 
         if (!m_ordered_pkg_id_list.empty())
         {
+            std::map<std::string, std::map<std::string, std::vector<PackageInfo>>> packageBuildsByVersion;
```

Comment:
Consider using `unordered_map` here instead of both usage of `map`, if that fits the usage (it seems to).

Side note, unrelated: we also have a `flat_set` (https://github.com/mamba-org/mamba/blob/main/libmamba/include/mamba/util/flat_set.hpp) in case you need a sorted vector that acts like a `std::set`. Might improve locality but I dont think that's applicable here, you decide, just wanted you to be aware we have that tool around 👍🏽

Reply:
Thanks for the pointer to the internal structure! As above, the packages already come ordered in `m_ordered_pkg_id_list` and to maintain that order a `map` is easier.